### PR TITLE
Mini Cart Contents: Refactor inner blocks styling using `currentColor`

### PR DESCRIPTION
**Pull request by dinhtungdu**
This PR refactors the style of Mini Cart Contents inner blocks to make them work better with color settings of Mini Cart Contents.

Note: This doesn't include style for quantity input which is shared between Cart and Mini Cart.

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests
### Manual Testing

How to test the changes in this Pull Request:

1. Edit the Mini Cart template part.
2. Set color and background color to custom values.
3. See the inner blocks applied correctly on both editor and front end.
### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
* [x] Same as above, or
* [ ] See steps below.

### CODE REVIEW

File: assets/js/blocks/cart-checkout/mini-cart/style.scss [hunk 1]
```diff
@@ -116,6 +116,7 @@ h2.wc-block-mini-cart__title {
 			}
 		}
 	}
+
```
Comment (Aljullu) [inline assets/js/blocks/cart-checkout/mini-cart/style.scss]:
```suggestion
```

File: assets/js/blocks/cart-checkout/mini-cart-contents/edit.tsx [hunk 2]
```diff
@@ -81,6 +82,30 @@ const Edit = ( { clientId }: Props ): ReactElement => {
 					templateLock={ false }
 				/>
 			</EditorProvider>
+			{ /**
+			 * This is a workaround to style inner blocks using the color
+			 * settings of the Mini Cart Contents block. It's possible to get
+			 * the Mini Cart Contents block's attributes inside the inner blocks
+			 * components, but we have 4/7 inner blocks that inherit style from
```
Comment (Aljullu) [inline assets/js/blocks/cart-checkout/mini-cart-contents/edit.tsx]: What does `4/7` mean here?

**Review by Aljullu — COMMENTED** [review]
I like how it simplifies the code, good job!

The only issue I noticed is that if I set the text color but not the background color, the Checkout button is not visible in the editor:

imatge

File: assets/js/blocks/cart-checkout/mini-cart-contents/edit.tsx (hunk 2 above)
Comment (dinhtungdu) [reply assets/js/blocks/cart-checkout/mini-cart-contents/edit.tsx]: it's 4 out of 7, I will update the comment to make it clearer.

**Comment by dinhtungdu** [comment]
> The only issue I noticed is that if I set the text color but not the background color, the Checkout button is not visible in the editor:

@Aljullu I can't reproduce this issue. I tried both Gutenberg and Core's Editor, the editor and front end.

image

I also tried setting only the background:

image

**Review by Aljullu — APPROVED** [review]
> @Aljullu I can't reproduce this issue.

Not sure what I did, I'm unable to reproduce again. Sorry for the false alarm. :sweat: 

This LGTM, so pre-approving. My only last concern is that it might print styles like `color: undefined` (when text color is not set). Do you think we could avoid this?

imatge

**Comment by dinhtungdu** [comment]
> My only last concern is that it might print styles like color: undefined (when text color is not set). Do you think we could avoid this?

@Aljullu Yes, we can. In 9ed5ec0, I check for the value of the style before returning it. I decided to create another component to handle the style there to keep the main edit component easy to read. The dedicated component is more future-proof because we may introduce new inner blocks (I may overthink here).

**Review by Aljullu — APPROVED** [review]
I like this abstraction! My only suggestion would be to rename the component from `InnerBlocksStyle` to `MiniCartInnerBlocksStyle`. The rationale being that currently, I don't think it can't be used in any other blocks/components, so I think it's better to make it clear in the component name that it's specific to the Mini Cart.

Another option would be to refactor it so things like the `selector` or the mappings (`color` → `style.backgroundColor`) are passed from the component props. That would pave the way in case sometime in the future we decide to use that component somewhere else. However, I don't think it's worth it until we need to do it. Thoughts?

**Comment by dinhtungdu** [comment]
> I like this abstraction! My only suggestion would be to rename the component from InnerBlocksStyle to MiniCartInnerBlocksStyle. 

@Aljullu I renamed the component per your suggestion.

> However, I don't think it's worth it until we need to do it. Thoughts?

I strongly agree. The first reason for the `MiniCartInnerBlocksStyle` is I want the edit component of the Mini Cart Contents block shorter and easier to read. When we actually need this for other blocks, we can refactor `MiniCartInnerBlocksStyle` at that point.

**Pull request merged**